src/context/ProfileContext.jsx

```diff
@@ -0,0 +1,44 @@
+import { createContext, useContext, useState } from 'react';
+import { useEffect } from 'react';
+import { Redirect } from 'react-router-dom';
+import { getProfile } from '../services/profile';
+
+export const ProfileContext = createContext();
```

Comment:
Context shouldn't really be necessary for the profile as it is not being used throughout the entire application. A custom hook handling this state that is called as needed may make more sense and avoid some of the bugs that come with your useEffect running before a user has a profile